pkg/admissioncontroller/webhooks/auth/seed/authorizer.go

```diff
@@ -110,6 +119,18 @@ func (a *authorizer) Authorize(_ context.Context, attrs auth.Attributes) (auth.D
 			return a.authorizeRead(seedName, graph.VertexTypeProject, attrs)
 		case secretBindingResource:
 			return a.authorizeRead(seedName, graph.VertexTypeSecretBinding, attrs)
+		case seedResource:
+			return a.authorize(seedName, graph.VertexTypeSeed, attrs,
+				[]string{"update", "patch", "delete"},
```

Comment:
I wonder if this would really correctly authorize "delete" here. To my knowledge, the only controller that deletes seeds is the ManagedSeed controller. However, it will never delete the seed it's gardenlet is responsible for (the "parent" seed), it will always try to delete "child" seeds. Even though such child seeds are always created by the "child" gardenlet, they are deleted by the "parent" instead as part of the reconciling the deletion of a ManagedSeed resource. Have you tested this?

I am not sure how to deal with this situation. Reworking gardenlet to delete its own seed seems pretty complicated, this is why it wasn't done like that in the first place. I am afraid we might need, at least temporarily, either allow "delete" of all seeds, or think about some more sophisticated "graph" logic to allow parent gardenlets to delete their children.

Reply:
Note: I just realized this can't be really tested, at least not with the current Nodeless `local-garden`. It requires a `remote-garden`, but with this setup `seed-authorizer` can't be properly enabled, see https://github.com/gardener/gardener/pull/3901#discussion_r617431845.

Reply:
Again, thanks for the very valuable feedback! As discussed via phone, with b1d139e I am now properly handling the cases where a "parent gardenlet" performs some requests for `Seed`s which it is not primarily responsible for, but which resulted out of a `ManagedSeed` reconciliation.

I will try to look into how to enable the `Seed{Authorizer,Restriction}` for the `remote-garden` setup when I'm done with the major implementation tasks.